### PR TITLE
Add keyvault authorizer

### PR DESCRIPTION
**Pull request by mcardosos**
Depends on https://github.com/Azure/go-autorest/pull/273

Thanks you for your contribution to the Azure-SDK-for-Go! We will triage and review it as quickly as we can.

As part of your submission, please make sure that you can make the following assertions:

 - [ ] I'm not making changes to Auto-Generated files which will just get erased next time there's a release.
   - If that's what you want to do, consider making a contribution here: https://github.com/Azure/autorest.go
 - [ ] I've tested my changes, adding unit tests where applicable.
 - [ ] I've added Apache 2.0 Headers to the top of any new source files.
 - [ ] I'm submitting this PR to the `dev` branch, or I'm fixing a bug that warrants its own release and I'm targeting the `master` branch.
 - [ ] If I'm targeting the `master` branch, I've also added a note to [CHANGELOG.md](https://github.com/Azure/azure-sdk-for-go/blob/master/README.md).
 - [ ] I've mentioned any relevant open issues in this PR, making clear the context for the contribution.

### CODE REVIEW

File: services/keyvault/auth/auth.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+)
+
+// NewAuthorizerFromEnvironment creates a keyvault dataplane Authorizer configured from environment variables in the order:
+// 1. Client credentials
+// 2. Client certificate
+// 3. Username password
+// 4. MSI
+func NewAuthorizerFromEnvironment() (autorest.Authorizer, error) {
```
Comment (vladbarosan) [inline services/keyvault/auth/auth.go]: changelog update and sdk version update ?

Comment (mcardosos) [reply services/keyvault/auth/auth.go]: Updated

**Comment by jhendrixMSFT** [comment]
Ugh yes, ideally the change would have been made in the latest branch (without the changelog/version update) then cherry-picked into master with the accompanying changelog/version.

**Comment by mcardosos** [comment]
Using PRs
https://github.com/Azure/azure-sdk-for-go/pull/1749
https://github.com/Azure/azure-sdk-for-go/pull/1752

**Pull request closed without merging**